### ISSUE TITLE
ART for TFLite models

### ISSUE THREAD

**Issue by Abhishek2271** [Trusted-AI/adversarial-robustness-toolbox#1212]
I am not sure if this is already supported. Does ART already support TFLite models for attack generation?

Current I get the error that 'byte' models are not reconized

**Comment by beat-buesser** [comment]
Hi @Abhishek2271 Thank you very much for your interest in ART! Could you please share here a short code example of how you are setting up the model with ART, to reproduce the error?

**Comment by Abhishek2271** [comment]
Hi,

Thank you for your reply. I have added my code snippet below. What I was doing was creating a TFLite model from Tensorflow and trying to use the TFLite model to create Adversarial examples via ART. Please find the code snippet below:

`
pip install -q tensorflow-model-optimization
import tempfile
import os
import tensorflow as tf
from tensorflow import keras
import tensorflow_model_optimization as tfmot


mnist = keras.datasets.mnist
(train_images, train_labels), (test_images, test_labels) = mnist.load_data()

# Normalize the input image so that each pixel value is between 0 to 1.
train_images = train_images / 255.0
test_images = test_images / 255.0

# Define the model architecture.
model = keras.Sequential([
  keras.layers.InputLayer(input_shape=(28, 28)),
  keras.layers.Reshape(target_shape=(28, 28, 1)),
  keras.layers.Conv2D(filters=12, kernel_size=(3, 3), activation='relu'),
  keras.layers.MaxPooling2D(pool_size=(2, 2)),
  keras.layers.Flatten(),
  keras.layers.Dense(10)
])


quantize_model = tfmot.quantization.keras.quantize_model
q_aware_model = quantize_model(model)
# `quantize_model` requires a recompile.
q_aware_model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])

q_aware_model.summary()

q_aware_model.fit(train_images_subset, train_labels_subset,
                  batch_size=500, epochs=1, validation_split=0.1)

converter = tf.lite.TFLiteConverter.from_keras_model(q_aware_model)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
tflite_model = converter.convert()

# Train the digit classification model
model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])

model.fit(
  train_images,
  train_labels,
  epochs=1,
  validation_split=0.1,
)


#Use art to create adversarial attacks
min_= 0.0
max_ =  1.0
loss_object = tf.keras.losses.SparseCategoricalCrossentropy()
classifier = TensorFlowV2Classifier(model=tflite_model, nb_classes=10, input_shape=(28, 28, 1), loss_object=loss_object, 
                                    clip_values=(0, 1), channels_first=False)
attack_fgsm = FastGradientMethod(estimator=classifier)
x_test_adv = attack_fgsm.generate(test_images)`

**Comment by beat-buesser** [comment]
Hi @Abhishek2271 The converter returns a binary string that can be written to a file in binary mode to store the model. I think to make inferences you would have to create first an Interpreter. These models can be attacked in general with ART's black-box classifier abstraction by providing a method the infers the model. For white-box attacks with `TensorFlowV2Classifier` the provided model has to be callable.

**Comment by Abhishek2271** [comment]
Hi @beat-buesser, thank you very much for the reply, I will try to black-box classifier but it seems that for now only one attack "HopSkipJump" is possible with the BlackBoxClassifier. 
Are other attacks like "Simple Black-box Adversarial Attack" which seems to be supported by ART as specified here: https://adversarial-robustness-toolbox.readthedocs.io/en/stable/modules/attacks/evasion.html#simple-black-box-adversarial-attack
possible via the Blackbox object? I will try it out but I could not find any examples for it on the guide.

**Comment by beat-buesser** [comment]
Hi @Abhishek2271 The `BlackBoxClassifier` and `BlackBoxClassifierNeuralNetwork` should be compatible with all 10 black-box attacks of ART: https://github.com/Trusted-AI/adversarial-robustness-toolbox/wiki/ART-Attacks#12-black-box

**Comment by Abhishek2271** [comment]
Oh I see it now, thank you.
